src/classes/USER_FrozenInActiveUser_TDTM.cls

```diff
@@ -0,0 +1,160 @@
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/**
+* @author Salesforce.org
+* @date 2017
+* @group User
+* @description Trigger handler on User object to handle User status changes.
+*/
+
+public class USER_FrozenInActiveUser_TDTM extends TDTM_Runnable {
+	
+	public USER_FrozenInActiveUser_TDTM() {}
+
+    /*******************************************************************************************************
+    * @description Trigger Handler on Opportunity that handles managing Opportunity Contact Roles
+    * @param listNew List of Opportunities from trigger new. 
+    * @param listOld List of Opportunities from trigger old. 
+    * @param triggerAction Which trigger event (BeforeInsert, AfterInsert, etc.). 
+    * @param objResult DescribeSObjectResult for Opportunity 
+    * @return dmlWrapper The objects to insert/update/delete
+    ********************************************************************************************************/
+    public override DmlWrapper run(List<SObject> newlist, List<SObject> oldlist,
+						              TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
+
+        DmlWrapper dmlWrapper = new DmlWrapper();
+
+        //================================
+        // Determine which handler method to call based on the triggerAction
+        //================================
+        if(TDTM_Runnable.Action.BeforeInsert == triggerAction) {
+            onBeforeInsert();
+
+        } else if(TDTM_Runnable.Action.BeforeUpdate == triggerAction) {
+            onBeforeUpdate();
+
+        } else if(TDTM_Runnable.Action.BeforeDelete == triggerAction) {
+            onBeforeDelete();
+
+        } else if(TDTM_Runnable.Action.AfterInsert == triggerAction) {
+            onAfterInsert();
+
+        } else if(TDTM_Runnable.Action.AfterUpdate == triggerAction) {
+            onAfterUpdate(newlist);
+
+        } else if(TDTM_Runnable.Action.AfterDelete == triggerAction) {
+            onAfterDelete();
+
+        } else if(TDTM_Runnable.Action.AfterUndelete == triggerAction) {
+            onAfterUnDelete();
+
+        }
+
```

Comment:
yes, I'd prefer this class to only refer to and implement the triggerAction for onAfterUpdate().  thanks.

Reply:
Sounds good. Thanks.